src/storage/store/node_column.cpp

```diff
@@ -186,19 +184,21 @@ void NodeColumn::scan(node_group_idx_t nodeGroupIdx, ColumnChunk* columnChunk) {
     } else {
         auto chunkMetadata = metadataDA->get(nodeGroupIdx, TransactionType::WRITE);
         auto cursor = PageElementCursor(chunkMetadata.pageIdx, 0);
-        auto numValuesToScan = chunkMetadata.numValues;
-        uint64_t numValuesScanned = 0;
-        while (numValuesScanned < numValuesToScan) {
-            uint64_t numValuesToScanInPage =
-                std::min((uint64_t)chunkMetadata.compMeta.numValues(
-                             BufferPoolConstants::PAGE_4KB_SIZE, dataType),
-                    numValuesToScan - numValuesScanned);
+        auto numValuesPerPage =
+            chunkMetadata.compMeta.numValues(BufferPoolConstants::PAGE_4KB_SIZE, dataType);
+        if (numValuesPerPage == UINT64_MAX) {
+            // This is a special case that bit_width is 0.
+            numValuesPerPage = 0;
+        }
```

Comment:
Looking forwards to implementing constant compression, I'm not sure this is the best way to handle this, as for constant compression we would have the same maximum values per page, but the values read would be non-zero, so we would actually need to scan them.

Then again, if the constant value is stored in the metadata, rather than in the page, then we would need a special way of scanning it anyway, but the simple implementation I would think (which would also work for variable sized values) would be to store it in the page once and copy it numValues times to the output.